AsyncDisplayKit/Layout/ASLayoutable.mm

```diff
@@ -38,3 +45,183 @@ struct ASLayoutableContext ASLayoutableGetCurrentContext()
 void ASLayoutableClearCurrentContext() {
   layoutableContextMap.erase(ASLayoutableGetCurrentContextKey());
 }
+
+/**
+ *  Given an id<ASLayoutable>, set up layout options that are intrinsically defined by the layoutable.
+ *
+ *  While this could be done in the layoutable object itself, moving the logic into this helper function
+ *  allows a custom spec to set up defaults without needing to alter the layoutable itself. For example,
+ *  image you were creating a custom baseline spec that needed ascender/descender. To assign values automatically
+ *  when a text node's attribute string is set, you would need to subclass ASTextNode and assign the values in the
+ *  override of setAttributeString. However, assigning the defaults via this function allows you to create a
+ *  custom spec without the need to create a subclass of ASTextNode.
+ *
+ *  @param layoutable The layoutable object to inspect for default intrinsic layout option values
+ */
+void ASLayoutableSetValuesForLayoutable(id<ASLayoutable> layoutable)
+{
```

Comment:
Move all of this logic in the concrete subclasses like ASDisplayNode, ASTextNode and add it to setPreferredFrameSize (ASDisplayNode) and setAttributedString (ASTextNode)